### PR TITLE
Update Rust runtime, add AsyncCallback

### PR DESCRIPTION
**Pull request by oowekyala**
This shows that lf-lang/reactor-rs#34 works properly. The test doesn't reproduce lf-lang/reactor-rs#33 though.

### CODE REVIEW

File: test/Rust/src/concurrent/AsyncCallback.lf [hunk 1]
```diff
@@ -0,0 +1,61 @@
+// Test asynchronous callbacks that trigger a physical action.
```
Comment (lhstrh) [inline test/Rust/src/concurrent/AsyncCallback.lf:1]: Probably have to run this through our code formatter `lff` to let this pass CI.

**Review by lhstrh — APPROVED** [review]

**Comment by lhstrh** [comment]
The latest Rust test failure looks a bit sus...

**Comment by oowekyala** [comment]
There is definitely something weird going on: https://github.com/lf-lang/lingua-franca/actions/runs/3429702634/jobs/5727270072#step:8:164
>  Reason: Test did not pass. Exit code: 139
>  This exit code typically indicates a segfault. In this case, the execution output is likely missing or incomplete.

I'm not sure how a segfault is possible given the test is safe Rust. I also would not expect the race condition to manifest like this. I think it would have been caught by the scheduler and cause a panic. I'll look into it

**Comment by oowekyala** [comment]
@lhstrh It looks like the new test is passing with the latest master. I still don't know why it used to segfault... maybe we can merge it anyway :shrug:

**Review by lhstrh — APPROVED** [review]

**Pull request merged**